Riot/Modules/TabBar/TabBarCoordinator.swift

```diff
@@ -699,8 +742,190 @@ final class TabBarCoordinator: NSObject, TabBarCoordinatorType {
         if let session = notification.object as? MXSession {
             showCoachMessageIfNeeded(with: session)
         }
+        
+        createAvatarButtonItem()
+    }
+    
+    private weak var rightMenuAvatarView: AvatarView?
+    
+    private func createAvatarButtonItem() {
+        guard BuildSettings.newAppLayoutEnaled else {
+            let searchBarButtonItem: MXKBarButtonItem = MXKBarButtonItem(image: Asset.Images.searchIcon.image, style: .plain) { [weak self] in
```

Comment:
Should this guard block do the same as below and check if the search bar button exists and return early if so? Can't remember if you said there was more work to do on this or not.

Reply:
I've made some changes to make things clearer.

Reply:
Definitely clearer now 🙌